src/message.rs

```diff
@@ -157,21 +157,53 @@ WHERE id=?;
         self.0
     }
 
+    /// Returns raw text of a message, used for message info
+    pub async fn rawtext(self, context: &Context) -> Result<String> {
+        Ok(context
+            .sql
+            .query_get_value("SELECT txt_raw FROM msgs WHERE id=?", (self,))
+            .await?
+            .unwrap_or_default())
+    }
+
+    /// Returns server foldernames and UIDs of a message, used for message info
+    pub async fn get_info_server_uids(
+        context: &Context,
+        rfc724_mid: String,
+    ) -> Result<Vec<(String, u32)>> {
+        context
+            .sql
+            .query_map(
+                "SELECT folder, uid FROM imap WHERE rfc724_mid=?",
+                (rfc724_mid,),
+                |row| {
+                    let folder: String = row.get("folder")?;
+                    let uid: u32 = row.get("uid")?;
+                    Ok((folder, uid))
+                },
+                |rows| {
+                    rows.collect::<std::result::Result<Vec<_>, _>>()
+                        .map_err(Into::into)
+                },
+            )
+            .await
+    }
+
+    /// Returns informtion about hops of a message, used for message info
```

Comment:
informAtion. And is Option really necessary here, maybe return an empty String or "No hop info" instead?

Reply:
> And is Option really necessary here, maybe return an empty String or "No hop info" instead?

I think it's fine in this case. ui could check for null and then not even display a hop info field or translate the "No hop info" string.
But we can still change this in the future anyway. I don't have a strong opinion about this.